samples/bluetooth/peripheral_gatt_dm/src/main.c

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <misc/printk.h>
+#include <misc/byteorder.h>
+#include <zephyr.h>
+#include <gpio.h>
+#include <soc.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+
+#include <bluetooth/gatt_dm.h>
+
+#define DEVICE_NAME             CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME_LEN         (sizeof(DEVICE_NAME) - 1)
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
+};
+
+static void discover_all_completed(struct bt_gatt_dm *dm, void *ctx)
+{
+	char uuid_str[37];
+
+	const struct bt_gatt_attr *gatt_service_attr =
+			bt_gatt_dm_service_get(dm);
+	const struct bt_gatt_service_val *gatt_service =
+			bt_gatt_dm_attr_service_val(gatt_service_attr);
+
+	size_t attr_count = bt_gatt_dm_attr_cnt(dm);
+
+	bt_uuid_to_str(gatt_service->uuid, uuid_str, sizeof(uuid_str));
+	printk("Found service %s\n", uuid_str);
+	printk("Attribute count: %d\n", attr_count);
+
+	bt_gatt_dm_data_print(dm);
```

Comment:
This should really be `data_log()` in my opinion

Reply:
What do you mean? Do you want me to change the name of `bt_gatt_dm_data_print`?

Reply:
never mind, not very important.